src/UI/home/dashboard/homeContext.tsx

```diff
@@ -1,30 +1,100 @@
-import React from "react";
-import {Chart} from "./chart";
+import React, { useEffect, useState } from 'react'
+import { Chart } from './chart'
+import { Wallet } from '../../../backend-service-connector/model/rest'
 
 export function HomeContext(props: any) {
-    const userService = props.userService
+    const walletService = props.walletService
```

Comment:
you can [desctructure](https://dmitripavlutin.com/javascript-object-destructuring/), so you don't have to write 'props.' over and over again

Reply:
is there a way to pass observer component like UserSrevice to all child without declare it in every single child? but only there where I need it. Or what i have done is correct?

Reply:
it's correct, but it uses [props drilling](https://www.geeksforgeeks.org/what-is-prop-drilling-and-how-to-avoid-it/). there are a few ways to avoid it. first of all, UserService is only a simple function so you can just use `import UserService from ...` in the required components and use it as const. another way (more abstract) is to use a global state handler like [context from react](https://beta.reactjs.org/learn/passing-data-deeply-with-context), or a third-party tool like [redux](https://redux.js.org/) or [zustand](https://github.com/pmndrs/zustand). imo the easiest is to use context because it is based on hooks that you are using throughout the app already. zustand is also pretty easy and has less boilerplate code. i recommend learning at least one tool for global state, it's required in almost all job offers

Reply:
no problem, I'm grateful that you want to help me, I'm improving my skills thanks to your help. i have a lot of questions to the app e.x. is better to use Class or Type to objects, can i get the accet to function from other functional component? beacuse i have other project and i use that what i learned with you and it makes a lot of complications with it, i have no idea how to convert all Class to function Component, 'observer' should be an class or could be function? I convert that chess project third time from half year 😂, first i do it use pure js with litte bit react, later i based on react components hooks and js class, nao i want to use that what i learned in this crypto project.